From 79c826c249001700007a6dffe80c3d61ff4428ef Mon Sep 17 00:00:00 2001
From: Sumit Garg <sumit.garg@linaro.org>
Date: Tue, 20 Jul 2021 13:54:30 +0530
Subject: [PATCH 1/2] plugins: Honour default cross compiler environment setup

Honour default $(CC), $(CFLAGS) and $(LDFLAGS) variables. Without this
plugins example fails to build for OE/Yocto.

Upstream-Status: Submitted [https://github.com/linaro-swg/optee_examples/pull/87]

Signed-off-by: Sumit Garg <sumit.garg@linaro.org>
---
 plugins/Makefile        |  2 +-
 plugins/host/Makefile   |  2 +-
 plugins/syslog/Makefile | 16 ++++++++++++----
 3 files changed, 14 insertions(+), 6 deletions(-)

diff --git a/plugins/Makefile b/plugins/Makefile
index 2372b38..ea472b4 100644
--- a/plugins/Makefile
+++ b/plugins/Makefile
@@ -8,7 +8,7 @@ TA_CROSS_COMPILE ?= $(CROSS_COMPILE)
 all:
 	$(MAKE) -C host CROSS_COMPILE="$(HOST_CROSS_COMPILE)" --no-builtin-variables
 	$(MAKE) -C ta CROSS_COMPILE="$(TA_CROSS_COMPILE)" LDFLAGS=""
-	$(MAKE) -C syslog CROSS_COMPILE="$(HOST_CROSS_COMPILE)"
+	$(MAKE) -C syslog CROSS_COMPILE="$(HOST_CROSS_COMPILE)" --no-builtin-variables
 
 .PHONY: clean
 clean:
diff --git a/plugins/host/Makefile b/plugins/host/Makefile
index 7285104..76244c7 100644
--- a/plugins/host/Makefile
+++ b/plugins/host/Makefile
@@ -20,7 +20,7 @@ BINARY = optee_example_plugins
 all: $(BINARY)
 
 $(BINARY): $(OBJS)
-	$(CC) -o $@ $< $(LDADD)
+	$(CC) $(LDFLAGS) -o $@ $< $(LDADD)
 
 .PHONY: clean
 clean:
diff --git a/plugins/syslog/Makefile b/plugins/syslog/Makefile
index 62d916a..71f5f92 100644
--- a/plugins/syslog/Makefile
+++ b/plugins/syslog/Makefile
@@ -1,3 +1,11 @@
+CC      ?= $(CROSS_COMPILE)gcc
+LD      ?= $(CROSS_COMPILE)ld
+AR      ?= $(CROSS_COMPILE)ar
+NM      ?= $(CROSS_COMPILE)nm
+OBJCOPY ?= $(CROSS_COMPILE)objcopy
+OBJDUMP ?= $(CROSS_COMPILE)objdump
+READELF ?= $(CROSS_COMPILE)readelf
+
 PLUGIN_UUID = 96bcf744-4f72-4866-bf1d-8634fd9c65e5
 
 PLUGIN			= $(PLUGIN_UUID).plugin
@@ -6,17 +14,17 @@ PLUGIN_OBJ		= $(patsubst %.c, %.o, $(PLUGIN_SRS))
 PLUGIN_INCLUDES_DIR	= $(CURDIR) $(TEEC_EXPORT)/include
 
 PLUGIN_INCLUDES		= $(addprefix -I, $(PLUGIN_INCLUDES_DIR))
-PLUGIN_CCFLAGS		= -Wall -fPIC
-PLUGIN_LDFLAGS		= -shared
+PLUGIN_CCFLAGS		= $(CFLAGS) -Wall -fPIC
+PLUGIN_LDFLAGS		= $(LDFLAGS) -shared
 
 .PHONY: all
 all: $(PLUGIN)
 
 $(PLUGIN): $(PLUGIN_OBJ)
-	$(CROSS_COMPILE)gcc $(PLUGIN_LDFLAGS) $(PLUGIN_OBJ) -o $@
+	$(CC) $(PLUGIN_LDFLAGS) $(PLUGIN_OBJ) -o $@
 
 %.o: %.c
-	$(CROSS_COMPILE)gcc $(PLUGIN_CCFLAGS) $(PLUGIN_INCLUDES) -c $*.c -o $*.o
+	$(CC) $(PLUGIN_CCFLAGS) $(PLUGIN_INCLUDES) -c $*.c -o $*.o
 
 .PHONY: clean
 clean:
-- 
2.25.1

